### PR TITLE
change: update utils library in command line tools.

### PR DESCRIPTION
**Pull request by github-zuoyan**
use gmatch deal http args

### CODE REVIEW

File: bin/utils/ngx_handle.lua [hunk 1]
```diff
@@ -9,12 +9,12 @@ local function create_dirs(necessary_dirs)
 end
 
 local function remove_ngx_conf(ngx_conf)
-    os.remove(ngx_conf)
+    os.command("mv -f "..ngx_conf .. " /tmp/")
```
Comment (shuaijinchao) [inline bin/utils/ngx_handle.lua]: why did  do this？

Comment (github-zuoyan) [reply bin/utils/ngx_handle.lua]: in OPS rules , linux command rm is forbidden.

File: bin/utils/command_util.lua [hunk 2]
```diff
@@ -26,7 +26,7 @@ function _M.run_command(command)
     local n = os.tmpname() -- get a temporary file name to store output
     local f = os.tmpname() -- get a temporary file name to store script
     _M.write_to_file(f, command)
-    local exit_code = os.execute("/bin/bash "..f.." > "..n.." 2>&1")
+    os.execute("/bin/bash "..f.." > "..n.." 2>&1")
```
[the remark this answers is not in the document]
Comment (github-zuoyan) [reply bin/utils/command_util.lua]: return the result  , but "exit_code" is unused. so I remove it from code

File: bin/utils/ngx_handle.lua (hunk 1 above)
Comment (shuaijinchao) [reply bin/utils/ngx_handle.lua]: can you give an example? os.remove is a Lua language package that is much safer than executing Linux commands directly.

Comment (github-zuoyan) [reply bin/utils/ngx_handle.lua]: I see it , "os.rename(ngx_conf, "/tmp/"..ngx_conf)" is better then "os.remove(ngx_conf)"

Comment (github-zuoyan) [reply bin/utils/ngx_handle.lua]: done

File: bin/utils/ngx_handle.lua [hunk 3]
```diff
@@ -8,25 +8,19 @@ local function create_dirs(necessary_dirs)
...
-    os.command("mv -f "..ngx_conf .. " /tmp/")
-end
 
 local function ngx_command(args)
     if not args then
         error("error args to execute nginx command.")
         os.exit(1)
     end
 
-    local orange_conf, prefix, ngx_conf, ngx_signal = "", "", "", ""
```
Comment (shuaijinchao) [inline bin/utils/ngx_handle.lua]: does this line of code need to be deleted? After making changes to the code, please submit it after local verification.

File: bin/utils/ngx_handle.lua (hunk 1 above)
Comment (shuaijinchao) [reply bin/utils/ngx_handle.lua]: OK

File: bin/utils/args_util.lua [hunk 4]
```diff
@@ -1,47 +1,20 @@
...
-    for match in (str .. delimiter):gmatch("(.-)" .. delimiter) do
-        table.insert(result, match)
-    end
-    return result
-end
-
 
 local _M = {}
 
 function _M.parse_args(args)
```
Comment (shuaijinchao) [inline bin/utils/args_util.lua]: I pull this PR back to the local test, this method did not implement the previous function, and throw an exception. Have you tested the function after modifying it?
我将这个PR拉回到本地进行测试，这个方法并没有实现之前的功能，并且抛出异常。请问你对函数进行修改后，进行了测试吗？

File: bin/utils/ngx_handle.lua [hunk 5]
```diff
@@ -8,25 +8,19 @@ local function create_dirs(necessary_dirs)
...
 
 local function ngx_command(args)
-    if not args then 
-        error("error args to execute nginx command.") 
+    if not args then
+        error("error args to execute nginx command.")
         os.exit(1)
     end
 
-    local orange_conf, prefix, ngx_conf, ngx_signal = "", "", "", ""
```
Comment (shuaijinchao) [inline bin/utils/ngx_handle.lua]: This line of code cannot be deleted, which makes local variables a global variable, which is not recommended.
这行代码不能删除，这会让局部变量成为全局变量，这是不推荐的。

File: bin/utils/args_util.lua (hunk 4 above)
Comment (shuaijinchao) [reply bin/utils/args_util.lua]: test the `parse_args` function throws the following exception.
对`parse_args`函数进行测试抛出了以下异常。
```bash
ERROR: ./bin/utils/args_util.lua:13: attempt to concatenate local 'args' (a table value)
stack traceback:
	./bin/main.lua:57: in function <./bin/main.lua:32>
	./bin/orange:38: in function 'file_gen'
	init_worker_by_lua:47: in function <init_worker_by_lua:45>
	[C]: in function 'xpcall'
	init_worker_by_lua:54: in function <init_worker_by_lua:52>
```

**Comment by shuaijinchao** [comment]
@github-zuoyan 
please modify the above two issues and submit them after the test.
请对以上两个问题进行修改，修改后请在本地测试成功后再进行提交。

**Comment by shuaijinchao** [comment]
hi, @github-zuoyan.
if you no longer fix the problem in the PR, I will close this PR today, if there are new ideas, welcome PR again.

**Comment by wujunze** [comment]
> hi, @github-zuoyan.
> if you no longer fix the problem in the PR, I will close this PR today, if there are new ideas, welcome PR again.

+1

**Pull request closed without merging**